### PR TITLE
Common opts

### LINKED ISSUE #80: -H and -P not working
while using the command import-item if i specify the host through the -H myhost.com it takes by default localhost so i have to always use --host!

### PR DESCRIPTION
**Pull request by santam85**
I did the following:

- factored out common options between various commands to a single place
- removed ambiguities with option aliases for watch and password
- refactored the way host configuration works for runtime and studio
- added forms shortcut overview command
- added support for other item types
- removed some antipatterns in the usage of promises
- fixed the cli silently failing in case of certain errors
- updated readmes and helps
- fixes #80 

plus other small fixes here and there.

### CODE REVIEW

**Comment by igord** [comment]
Changing common shortcuts is a breaking change. I am not sure that it is a good idea to do it now.

**Comment by santam85** [comment]
There was a conflict for some, had the same shortcut, the long version
stayed the same, moreover i factored the common one out to a single place
so we can change them uniformly. Only one changed is password

**Comment by santam85** [comment]
Hey @igord , I reverted the shortcuts names, this can be merged now.

File: commands/import-item.js [hunk 1]
```diff
@@ -43,66 +43,59 @@ module.exports = Command.extend({
         return r;
     },
 
-    options: {
+    options: util.buildOpts({
         target: {type: 'string', alias: 't', default: './'},
-        watch: {type: 'boolean', alias: 'W'},
+        watch: {type: 'boolean', alias: 'w'},
```
Comment (Emigre) [inline commands/import-item.js]: Is this correct? The prompt at the top of the file says that -w is for --password, and -W for --watch

Comment (santam85) [inline commands/import-item.js]: Good spot, I'll fix it

**Pull request merged**